src/t_cose_encrypt_dec.c

```diff
@@ -193,6 +193,16 @@ t_cose_encrypt_dec_detached(struct t_cose_encrypt_dec_ctx* me,
 
     nonce_cbor = t_cose_param_find_iv(body_params_list);
     ce_alg.cose_alg_id = t_cose_param_find_alg_id_prot(body_params_list);
+    if(ce_alg.cose_alg_id == T_COSE_ALGORITHM_NONE) {
```

Comment:
Can use t_cose_param_find_alg_id() here instead. It will return the protection and you can check it.

The problem is there is still a check needed that there are no other protected headers. I guess you just iterate over the linked list. Was hoping to use t_cose_headers_decode(,,true,,,) for that, but it doesn't work.